dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/Switch.extvar

```diff
@@ -0,0 +1,18 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!--
+    Copyright (c) 2022, RTE (http://www.rte-france.com)
+    See AUTHORS.txt
+    All rights reserved.
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, you can obtain one at http://mozilla.org/MPL/2.0/.
+    SPDX-License-Identifier: MPL-2.0
+
+    This file is part of Dynawo, an hybrid C++/Modelica open source suite of
+    simulation tools for power systems.
+-->
+<external_variables xmlns="http://www.rte-france.com/dynawo">
+  <variable id="u1" type="discrete"/>
```

Comment:
Are u1 and u2 necessary discrete in your case ?

Reply:
Yes! 
The documentation / comments in the .mo have been modified to make it more obvious.